### ISSUE TITLE
JSPs not supported.

### ISSUE THREAD

**Issue by VasilMitovClouway** [GoogleCloudPlatform/gradle-appengine-plugin#276]
I am not sure that this is a bug, or it is supposed to act this way and I'm just missing something. I was following the tutorial and tried to upload a single JSP file with gradle appengineUpdate and although all of the settings for JDK and source code target are set to be to 7/1.7, JSPs are not translated (i'm not sure im using the correct term) properly and throw java.lang.IllegalArgumentException: Class file is Java 8 but max supported is Java 7. 

![screenshot from 2017-02-21 09-57-27](https://cloud.githubusercontent.com/assets/18255988/23155707/75cf1c24-f81c-11e6-9562-bc1ec56b256c.png)

Project works fine and deploys without a problem when the JSPs are removed and Servlets are used.

**Comment by loosebazooka** [comment]
The JDK is set to Java7? The jsp compiler is loaded by the underlying tooling, if you use a different Java version (java sdk 8) and set the target version (java 7), I'm not sure what the behavior for jsp compilation, but maybe it's just using java8.

**Comment by VasilMitovClouway** [comment]
Nah, I particularly changed the sdk from 8 to 7, I even changed the default Java path environmental variable to use Java 1.7. I was stuck on this problem for along time and I am pretty sure I tried every possible way/setting I could think of.

**Comment by patflynn** [comment]
@loosebazooka could be worth seeing if we can reproduce.  This looks similar to an earlier Java 8 related SDK bug.
